% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/io.R
\name{extract_counts}
\alias{extract_counts}
\title{Extract and store data needed to compute blocked f2}
\usage{
extract_counts(
  pref,
  outdir,
  inds = NULL,
  blgsize = 0.05,
  maxmiss = 0,
  minmaf = 0,
  maxmaf = 0.5,
  transitions = TRUE,
  transversions = TRUE,
  auto_only = TRUE,
  keepsnps = NULL,
  maxmem = 8000,
  overwrite = FALSE,
  format = NULL,
  cols_per_chunk = NULL,
  verbose = TRUE
)
}
\arguments{
\item{pref}{Prefix of \emph{PLINK/EIGENSTRAT/PACKEDANCESTRYMAP} files.
\emph{EIGENSTRAT/PACKEDANCESTRYMAP} have to end in \code{.geno}, \code{.snp}, \code{.ind}, \emph{PLINK} has to end in \code{.bed}, \code{.bim}, \code{.fam}}

\item{outdir}{Directory where data will be stored.}

\item{inds}{Individuals for which data should be read. Defaults to all individuals}

\item{blgsize}{SNP block size in Morgan. Default is 0.05 (50 cM). If \code{blgsize} is 100 or greater, if will be interpreted as base pair distance rather than centimorgan distance.}

\item{maxmiss}{Discard SNPs which are missing in a fraction of individuals greater than \code{maxmiss}}

\item{minmaf}{Discard SNPs with minor allele frequency less than \code{minmaf}}

\item{maxmaf}{Discard SNPs with minor allele frequency greater than than \code{maxmaf}}

\item{transitions}{Set this to \code{FALSE} to exclude transition SNPs}

\item{transversions}{Set this to \code{FALSE} to exclude transversion SNPs}

\item{auto_only}{Keep only SNPs on chromosomes 1 to 22}

\item{keepsnps}{SNP IDs of SNPs to keep. Overrides other SNP filtering options}

\item{maxmem}{Maximum amount of memory to be used. If the required amount of memory exceeds \code{maxmem}, allele frequency data will be split into blocks, and the computation will be performed separately on each block pair. This doesn't put a precise cap on the amount of memory used (it used to at some point). Set this parameter to lower values if you run out of memory while running this function. Set it to higher values if this function is too slow and you have lots of memory.}

\item{overwrite}{Overwrite existing files in \code{outdir}}

\item{format}{Supply this if the prefix can refer to genotype data in different formats
and you want to choose which one to read. Should be \code{plink} to read \code{.bed}, \code{.bim}, \code{.fam} files, or \code{eigenstrat}, or \code{packedancestrymap} to read \code{.geno}, \code{.snp}, \code{.ind} files.}

\item{cols_per_chunk}{Number of genotype chunks to store on disk. Setting this to a positive integer makes the function slower, but requires less memory. The default value for \code{cols_per_chunk} in \code{\link{extract_afs}} is 10. Lower numbers will lower the memory requirement but increase the time it takes.}

\item{verbose}{Print progress updates}
}
\description{
Prepare data for various \emph{ADMIXTOOLS 2} functions. This function reads data from genotype files,
and extracts data required to compute blocked f-statistics for any sets of samples. The data consists of
\code{.rds} files with total and alternative allele counts for each individual, and products of total
and alternative allele counts for each pair.
The function calls \code{\link{packedancestrymap_to_afs}} or \code{\link{plink_to_afs}}
and \code{\link{afs_to_f2_blocks}}.
}
