<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data formats â€¢ admixtools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><script src="../extra.js"></script><meta property="og:title" content="Data formats">
<meta property="og:description" content="admixtools">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-9DRRL9PZD0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-9DRRL9PZD0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">admixtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/admixtools.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/fstats.html">f-statistics</a>
    </li>
    <li>
      <a href="../articles/graphs.html">Admixture graphs</a>
    </li>
    <li>
      <a href="../articles/io.html">Data formats</a>
    </li>
    <li>
      <a href="../articles/paper.html">Paper notes</a>
    </li>
    <li>
      <a href="../articles/parallel.html">Parallelization</a>
    </li>
    <li>
      <a href="../articles/plotting.html">Plotting</a>
    </li>
    <li>
      <a href="../articles/qpadm.html">qpWave and qpAdm</a>
    </li>
    <li>
      <a href="../articles/resampling.html">Standard errors</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/uqrmaie1/admixtools/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="io_files/header-attrs-2.9/header-attrs.js"></script><script src="io_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="io_files/plotly-binding-4.9.4.1/plotly.js"></script><script src="io_files/typedarray-0.1/typedarray.min.js"></script><script src="io_files/jquery-3.5.1/jquery.min.js"></script><link href="io_files/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="io_files/crosstalk-1.1.1/js/crosstalk.min.js"></script><link href="io_files/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="io_files/plotly-main-1.57.1/plotly-latest.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data formats</h1>
                        <h4 class="author">Robert Maier</h4>
            
            <h4 class="date">2022-11-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/uqrmaie1/admixtools/blob/master/vignettes/io.Rmd"><code>vignettes/io.Rmd</code></a></small>
      <div class="hidden name"><code>io.Rmd</code></div>

    </div>

    
    
<div id="genotype-data-formats" class="section level2">
<h2 class="hasAnchor">
<a href="#genotype-data-formats" class="anchor"></a>Genotype data formats</h2>
<p><em>ADMIXTOOLS 2</em> can read genotype data in three formats.</p>
<ol style="list-style-type: decimal">
<li>Binary <em>PLINK</em> format (PACKEDPED), described <a href="https://www.cog-genomics.org/plink2/input">here</a>
</li>
<li>Binary <em>PACKEDANCESTRYMAP</em> format</li>
<li>Text based <em>EIGENSTRAT</em> format</li>
</ol>
<p><em>PACKEDANCESTRYMAP</em> and <em>EIGENSTRAT</em> are described <a href="https://reich.hms.harvard.edu/software/InputFileFormats">here</a></p>
<p>In all three formats a dataset consists of three files: one file for the genotype matrix, one file for the SNP metadata, and one file for the sample metadata.</p>
<p>The following shows what a data set for three SNPs and five samples grouped into two populations (French.DG, Onge.DG) could look like:</p>
<div id="plink" class="section level3">
<h3 class="hasAnchor">
<a href="#plink" class="anchor"></a>PLINK</h3>
<p>PLINK <code>.fam</code> files, have six columns, but only the first two (family/population ID and sample ID) are relevant in <em>ADMIXTOOLS 2</em></p>
<pre><code>French.DG   S_French-1.DG   0   0   M   -9
French.DG   S_French-2.DG   0   0   F   -9
French.DG   B_French-3.DG   0   0   M   -9
Onge.DG BR_Onge-2.DG    0   0   F   -9
Onge.DG BR_Onge-1.DG    0   0   F   -9</code></pre>
<p>PLINK <code>.bim</code> file:</p>
<pre><code>1   rs3094315   0.02013 752566  G   A
1   rs12124819  0.020242    776546  A   G
1   rs28765502  0.022137    832918  T   C</code></pre>
</div>
<div id="eigenstratpackedancestrymap" class="section level3">
<h3 class="hasAnchor">
<a href="#eigenstratpackedancestrymap" class="anchor"></a>EIGENSTRAT/PACKEDANCESTRYMAP</h3>
<p><em>EIGENSTRAT</em> and <em>PACKEDANCESTRYMAP</em> use the same format for the metadata. The only difference is that the genotype data is stored in a text file in <em>EIGENSTRAT</em>, but in a binary file in <em>PACKEDANCESTRYMAP</em>.</p>
<p>The <code>.ind</code> file has 3 out of the 6 columns in the <code>.fam</code> file, in different order:</p>
<pre><code>S_French-1.DG M French.DG
S_French-2.DG F French.DG
B_French-3.DG M French.DG
BR_Onge-2.DG F Onge.DG
BR_Onge-1.DG F Onge.DG</code></pre>
<p><code>.snp</code> files are the same as <code>.bim</code> files, except that the first two columns are swapped:</p>
<pre><code>rs3094315 1 0.02013 752566  G   A
rs12124819 1    0.020242    776546  A   G
rs28765502 1    0.022137    832918  T   C</code></pre>
<p><em>EIGENSTRAT</em> <code>.geno</code> files look like this:</p>
<pre><code>29101
10211
22901</code></pre>
<p>One SNP per row, one sample per column, and missing data is denoted by <code>9</code>.</p>
</div>
</div>
<div id="reading-genotype-files" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-genotype-files" class="anchor"></a>Reading genotype files</h2>
<p>While there are other R packages for reading and writing binary <em>PLINK</em> files (<a href="https://github.com/gabraham/plink2R">plink2R</a>, <a href="https://github.com/OchoaLab/genio">genio</a>), no such packages exist to date for <em>PACKEDANCESTRYMAP</em> format files, so some of the functions shown here may be useful outside of the <em>ADMIXTOOLS</em> framework.</p>
<p>Note that the functions below impose the restriction than individual IDs may not be duplicated across populations or families.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">geno</span> <span class="op">=</span> <span class="fu"><a href="../reference/read_plink.html">read_plink</a></span><span class="op">(</span><span class="st">"plink/prefix"</span><span class="op">)</span>
<span class="va">geno</span> <span class="op">=</span> <span class="fu"><a href="../reference/read_eigenstrat.html">read_eigenstrat</a></span><span class="op">(</span><span class="st">"eigenstrat/prefix"</span><span class="op">)</span>
<span class="va">geno</span> <span class="op">=</span> <span class="fu"><a href="../reference/read_packedancestrymap.html">read_packedancestrymap</a></span><span class="op">(</span><span class="st">"packedancestrymap/prefix"</span><span class="op">)</span>

<span class="co"># read SNPs 1000 to 2000</span>
<span class="va">geno</span> <span class="op">=</span> <span class="fu"><a href="../reference/read_packedancestrymap.html">read_packedancestrymap</a></span><span class="op">(</span><span class="st">"packedancestrymap/prefix"</span>, first <span class="op">=</span> <span class="fl">1000</span>, last <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span>
<span class="co"># read only ind1 and ind2</span>
<span class="va">geno</span> <span class="op">=</span> <span class="fu"><a href="../reference/read_packedancestrymap.html">read_packedancestrymap</a></span><span class="op">(</span><span class="st">"packedancestrymap/prefix"</span>, inds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ind1'</span>, <span class="st">'ind2'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="extracting-populations" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting-populations" class="anchor"></a>Extracting populations</h2>
<p>Populations can be extracted from <em>PLINK</em> files (and written to another set of <em>PLINK</em> files) like this:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/extract_samples.html">extract_samples</a></span><span class="op">(</span><span class="st">"input/prefix"</span>, <span class="st">"output/prefix"</span>, pops <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"pop1"</span>, <span class="st">"pop2"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><code>extract_samples</code> can also be used to extract samples, similar to <code>plink --keep</code>.</p>
</div>
<div id="combining-data-sets" class="section level2">
<h2 class="hasAnchor">
<a href="#combining-data-sets" class="anchor"></a>Combining data sets</h2>
<p><em>ADMIXTOOLS 2</em> canâ€™t merge data sets. Data in <em>PLINK</em> format can be merged using <code>plink --bmerge</code>.</p>
</div>
<div id="computing-allele-frequencies" class="section level2">
<h2 class="hasAnchor">
<a href="#computing-allele-frequencies" class="anchor"></a>Computing allele frequencies</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">afs</span> <span class="op">=</span> <span class="fu"><a href="../reference/plink_to_afs.html">plink_to_afs</a></span><span class="op">(</span><span class="st">"plink/prefix"</span><span class="op">)</span>
<span class="va">afs</span> <span class="op">=</span> <span class="fu"><a href="../reference/eigenstrat_to_afs.html">eigenstrat_to_afs</a></span><span class="op">(</span><span class="st">"eigenstrat/prefix"</span><span class="op">)</span>
<span class="va">afs</span> <span class="op">=</span> <span class="fu"><a href="../reference/packedancestrymap_to_afs.html">packedancestrymap_to_afs</a></span><span class="op">(</span><span class="st">"packedancestrymap/prefix"</span><span class="op">)</span></code></pre></div>
</div>
<div id="converting-packedancestrymap-to-plink" class="section level2">
<h2 class="hasAnchor">
<a href="#converting-packedancestrymap-to-plink" class="anchor"></a>Converting <em>PACKEDANCESTRYMAP</em> to <em>PLINK</em>
</h2>
<p>This function converts genotype files from <em>PACKEDANCESTRYMAP</em> or <em>EIGENSTRAT</em> to <em>PLINK</em> format. If <code>inds</code> or <code>pops</code> is specified, only those samples will be extracted.</p>
<p><em>ADMIXTOOLS 2</em> currently doesnâ€™t provide the option to write binary <em>PACKEDANCESTRYMAP</em> files, so the conversion only goes one way.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/packedancestrymap_to_plink.html">eigenstrat_to_plink</a></span><span class="op">(</span><span class="st">"eigenstrat_input_prefix"</span>, <span class="st">"plink_output_prefix"</span><span class="op">)</span>
<span class="fu"><a href="../reference/packedancestrymap_to_plink.html">packedancestrymap_to_plink</a></span><span class="op">(</span><span class="st">"packedancestrymap_input_prefix"</span>, <span class="st">"plink_output_prefix"</span><span class="op">)</span></code></pre></div>
</div>
<div id="admixture-graph-formats" class="section level2">
<h2 class="hasAnchor">
<a href="#admixture-graph-formats" class="anchor"></a>Admixture graph formats</h2>
<p>There are several ways in which admixture graphs can be represented. Some of them are useful because they can be stored as human readable text files, and others are useful because they make it easier for R to process graphs. <em>ADMIXTOOLS 2</em> has several function for importing, exporting and converting between formats.</p>
<p>Before fitting an admixture graph, the graph only has nodes and edges. After fitting a graph, each edge has a weight associated with it, and maybe some other information. Here I mostly show examples of graphs that are not fitted yet.</p>
<div id="edge-list-format" class="section level3">
<h3 class="hasAnchor">
<a href="#edge-list-format" class="anchor"></a>Edge list format</h3>
<p>The simplest way to represent a graph in a text file is by writing each edge in the graph in a separate line:</p>
<pre><code>R      O
R      n1
n1     n2
n1     n3
n2     A
n2     admix
n3     B
n3     n4
n4     C
n4     admix
admix  D</code></pre>
<p>We can save this in a text file <code>graph.txt</code> and read it into R like this:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">graph_el</span> <span class="op">=</span> <span class="fu">read_table2</span><span class="op">(</span><span class="st">'~/Downloads/graph.tsv'</span>, col_names <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotly_graph.html">plotly_graph</a></span><span class="op">(</span><span class="va">graph_el</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-2a69bef99ceda2d8a8f3" style="width:700px;height:432.632880098888px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-2a69bef99ceda2d8a8f3">{"x":{"data":[{"x":[2,0],"y":[1,0],"text":"n4 -> admix","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(163,165,0,1)","dash":"dot"},"hoveron":"points","name":"(admix,1)","legendgroup":"(admix,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.5,0],"y":[2,0],"text":"n2 -> admix","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(0,191,125,1)","dash":"dot"},"hoveron":"points","name":"(admix,2)","legendgroup":"(admix,2)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,0],"y":[0,-1],"text":"admix -> D","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(248,118,109,1)","dash":"solid"},"hoveron":"points","name":"(normal,0)","legendgroup":"(normal,0)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2,2],"y":[1,0],"text":"n4 -> C","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(163,165,0,1)","dash":"solid"},"hoveron":"points","name":"(normal,1)","legendgroup":"(normal,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.5,-1,null,1.5,1,null,1.5,2],"y":[2,1,null,2,1,null,2,1],"text":["n2 -> A","n2 -> A",null,"n3 -> B","n3 -> B",null,"n3 -> n4","n3 -> n4"],"type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(0,191,125,1)","dash":"solid"},"hoveron":"points","name":"(normal,2)","legendgroup":"(normal,2)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.5,-0.5,null,0.5,1.5],"y":[3,2,null,3,2],"text":["n1 -> n2","n1 -> n2",null,"n1 -> n3","n1 -> n3"],"type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(0,176,246,1)","dash":"solid"},"hoveron":"points","name":"(normal,3)","legendgroup":"(normal,3)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,-0.5,null,0,0.5],"y":[4,3,null,4,3],"text":["R -> O","R -> O",null,"R -> n1","R -> n1"],"type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(231,107,243,1)","dash":"solid"},"hoveron":"points","name":"(normal,4)","legendgroup":"(normal,4)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,0.5,-0.5,1.5,2,0],"y":[4,3,2,2,1,0],"text":["R","n1","n2","n3","n4","admix"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,0,0,1)","opacity":0,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.25,0.25,0,1,-0.75,-0.25,1.25,1.75,2,1,0],"y":[3.5,3.5,2.5,2.5,1.5,1,1.5,1.5,0.5,0.5,-0.5],"text":["","","","","","","","","","",""],"hovertext":["R -> O","R -> n1","n1 -> n2","n1 -> n3","n2 -> A","n2 -> admix","n3 -> B","n3 -> n4","n4 -> C","n4 -> admix","admix -> D"],"textfont":{"size":9.4488188976378,"color":"rgba(0,0,0,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0],"y":[-1.1],"text":"D","hovertext":"","textfont":{"size":9.4488188976378,"color":"rgba(248,118,109,1)"},"type":"scatter","mode":"text","hoveron":"points","name":"(0,1)","legendgroup":"(0,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2],"y":[-0.1],"text":"C","hovertext":"","textfont":{"size":9.4488188976378,"color":"rgba(163,165,0,1)"},"type":"scatter","mode":"text","hoveron":"points","name":"1","legendgroup":"1","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-1,1],"y":[0.9,0.9],"text":["A","B"],"hovertext":["",""],"textfont":{"size":9.4488188976378,"color":"rgba(0,191,125,1)"},"type":"scatter","mode":"text","hoveron":"points","name":"(2,1)","legendgroup":"(2,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.5],"y":[2.9],"text":"O","hovertext":"","textfont":{"size":9.4488188976378,"color":"rgba(231,107,243,1)"},"type":"scatter","mode":"text","hoveron":"points","name":"(4,1)","legendgroup":"(4,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"visible":false,"showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.5],"y":[4],"text":"","hovertext":"","textfont":{"size":14.6645669291339,"color":"rgba(0,0,0,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":27.8246364188244,"r":7.30593607305936,"b":15.4776044553541,"l":10.958904109589},"paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1.4,2.4],"tickmode":"array","ticktext":["-1","0","1","2"],"tickvals":[-1,0,1,2],"categoryorder":"array","categoryarray":["-1","0","1","2"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":{"text":"","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1.355,4.255],"tickmode":"array","ticktext":["-1","0","1","2","3","4"],"tickvals":[-1,0,1,2,3,4],"categoryorder":"array","categoryarray":["-1","0","1","2","3","4"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","title":{"text":"","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"76f17051877":{"x":{},"y":{},"xend":{},"yend":{},"from":{},"to":{},"linetype":{},"colour":{},"text":{},"type":"scatter"},"76f15f89303a":{"x":{},"y":{},"xend":{},"yend":{},"from":{},"to":{},"text":{}},"76f127bd60d6":{"x":{},"y":{},"xend":{},"yend":{},"from":{},"to":{},"label":{},"text":{}},"76f122095fc":{"x":{},"y":{},"xend":{},"yend":{},"from":{},"to":{},"label":{},"colour":{}},"76f12956d47b":{"y":{}},"76f1397ec71":{"x":{},"y":{}}},"cur_data":"76f17051877","visdat":{"76f17051877":["function (y) ","x"],"76f15f89303a":["function (y) ","x"],"76f127bd60d6":["function (y) ","x"],"76f122095fc":["function (y) ","x"],"76f12956d47b":["function (y) ","x"],"76f1397ec71":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script><p><code><a href="../reference/qpgraph.html">qpgraph()</a></code> uses this format to represent fitted graphs. In addition to the first two columns (<code>from</code> and <code>to</code>), there will be additional columns to indicate the type of edge, its estimated weight, and possibly more information.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/qpgraph.html">qpgraph</a></span><span class="op">(</span><span class="va">example_f2_blocks</span>, <span class="va">example_igraph</span><span class="op">)</span><span class="op">$</span><span class="va">edges</span> <span class="op">%&gt;%</span> <span class="va">head</span></code></pre></div>
<pre><code>## # A tibble: 6 x 6
##   from  to                   type   weight   low  high
##   &lt;chr&gt; &lt;chr&gt;                &lt;chr&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1 R     Chimp.REF            edge  0.0724     NA    NA
## 2 R     N4N                  edge  0.0724     NA    NA
## 3 N0N   Altai_Neanderthal.DG edge  0.00321    NA    NA
## 4 N0N   N3N0                 edge  0.0111     NA    NA
## 5 N1N   N0N                  edge  0.0841     NA    NA
## 6 N1N   N3N1                 edge  0.0362     NA    NA</code></pre>
</div>
<div id="igraph-format" class="section level3">
<h3 class="hasAnchor">
<a href="#igraph-format" class="anchor"></a>igraph format</h3>
<p>Many <em>ADMIXTOOLS 2</em> functions use the <code>igraph</code> package, which has its own graph format. The most important <em>ADMIXTOOLS 2</em> functions work with graphs in either the edge list or the <code>igraph</code> format, but some only work with graphs in the <code>igraph</code> format. Itâ€™s easy to convert between the two formats:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">graph_el</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "tbl_df"     "tbl"        "data.frame"</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">graph</span> <span class="op">=</span> <span class="fu"><a href="../reference/edges_to_igraph.html">edges_to_igraph</a></span><span class="op">(</span><span class="va">graph_el</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "igraph"</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">graph_el2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/as_edgelist.html">as_edgelist</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">graph_el2</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "data.frame"</code></pre>
</div>
<div id="original-admixtools-format" class="section level3">
<h3 class="hasAnchor">
<a href="#original-admixtools-format" class="anchor"></a>Original ADMIXTOOLS format</h3>
<p>The text representation of the graph above in the original <em>ADMIXTOOLS</em> software would look like this:</p>
<pre><code>vertex              R
vertex              O
vertex              n1
vertex              n2
vertex              n3
vertex              A
vertex              admix
vertex              B
vertex              n4
vertex              C
vertex              D
label               O                   O
label               A                   A
label               B                   B
label               C                   C
label               D                   D
edge                R_O                 R                   O
edge                R_n1                R                   n1
edge                n1_n2               n1                  n2
edge                n1_n3               n1                  n3
edge                n2_A                n2                  A
edge                n3_B                n3                  B
edge                n3_n4               n3                  n4
edge                n4_C                n4                  C
edge                admix_D             admix               D
admix               admix               n2                  n4</code></pre>
<p>If this is stored in a file called <code>graph.txt</code>, it can be imported into R like this:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">graph_el</span> <span class="op">=</span> <span class="fu"><a href="../reference/parse_qpgraph_graphfile.html">parse_qpgraph_graphfile</a></span><span class="op">(</span><span class="st">'graph.txt'</span><span class="op">)</span></code></pre></div>
</div>
<div id="dot-format" class="section level3">
<h3 class="hasAnchor">
<a href="#dot-format" class="anchor"></a>DOT format</h3>
<p>DOT format is often used for plotting admixture graphs. You can export a fitted admixture graph like this:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="../reference/qpgraph.html">qpgraph</a></span><span class="op">(</span><span class="va">example_f2_blocks</span>, <span class="va">example_igraph</span><span class="op">)</span><span class="op">$</span><span class="va">edges</span>
<span class="fu"><a href="../reference/write_dot.html">write_dot</a></span><span class="op">(</span><span class="va">fit</span>, <span class="st">'graph.dot'</span><span class="op">)</span></code></pre></div>
<p><em>ADMIXTOOLS 2</em> currently doesnâ€™t have a function for reading DOT files.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robert Maier, Nick Patterson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'c30b6be96312414d83a0ba52b94b558b',
    indexName: 'admixtools',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
