<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Compute and store blocked f2 statistics — extract_f2_large • admixtools</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


  
  <script src="../extra.js"></script>

<meta property="og:title" content="Compute and store blocked f2 statistics — extract_f2_large" />
<meta property="og:description" content="extract_f2_large does the same as extract_f2, but it requires less memory and is slower. outdir has to be set in extract_f2_large." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-9DRRL9PZD0"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-9DRRL9PZD0');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">admixtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/admixtools.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/fstats.html">f-statistics</a>
    </li>
    <li>
      <a href="../articles/graphs.html">Admixture graphs</a>
    </li>
    <li>
      <a href="../articles/io.html">Data formats</a>
    </li>
    <li>
      <a href="../articles/paper.html">Paper notes</a>
    </li>
    <li>
      <a href="../articles/parallel.html">Parallelization</a>
    </li>
    <li>
      <a href="../articles/plotting.html">Plotting</a>
    </li>
    <li>
      <a href="../articles/qpadm.html">qpWave and qpAdm</a>
    </li>
    <li>
      <a href="../articles/resampling.html">Standard errors</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/uqrmaie1/admixtools/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compute and store blocked f2 statistics</h1>
    <small class="dont-index">Source: <a href='https://github.com/uqrmaie1/admixtools/blob/master/R/io.R'><code>R/io.R</code></a></small>
    <div class="hidden name"><code>extract_f2_large.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>extract_f2_large</code> does the same as <code><a href='extract_f2.html'>extract_f2</a></code>, but it requires less memory and is slower. <code>outdir</code> has to be set in <code>extract_f2_large</code>.</p>
    </div>

    <pre class="usage"><span class='fu'>extract_f2_large</span><span class='op'>(</span>
  <span class='va'>pref</span>,
  <span class='va'>outdir</span>,
  inds <span class='op'>=</span> <span class='cn'>NULL</span>,
  pops <span class='op'>=</span> <span class='cn'>NULL</span>,
  blgsize <span class='op'>=</span> <span class='fl'>0.05</span>,
  cols_per_chunk <span class='op'>=</span> <span class='fl'>10</span>,
  maxmiss <span class='op'>=</span> <span class='fl'>0</span>,
  minmaf <span class='op'>=</span> <span class='fl'>0</span>,
  maxmaf <span class='op'>=</span> <span class='fl'>0.5</span>,
  minac2 <span class='op'>=</span> <span class='cn'>FALSE</span>,
  outpop <span class='op'>=</span> <span class='cn'>NULL</span>,
  outpop_scale <span class='op'>=</span> <span class='cn'>TRUE</span>,
  transitions <span class='op'>=</span> <span class='cn'>TRUE</span>,
  transversions <span class='op'>=</span> <span class='cn'>TRUE</span>,
  keepsnps <span class='op'>=</span> <span class='cn'>NULL</span>,
  snpblocks <span class='op'>=</span> <span class='cn'>NULL</span>,
  overwrite <span class='op'>=</span> <span class='cn'>FALSE</span>,
  format <span class='op'>=</span> <span class='cn'>NULL</span>,
  adjust_pseudohaploid <span class='op'>=</span> <span class='cn'>TRUE</span>,
  afprod <span class='op'>=</span> <span class='cn'>TRUE</span>,
  fst <span class='op'>=</span> <span class='cn'>TRUE</span>,
  poly_only <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"f2"</span><span class='op'>)</span>,
  apply_corr <span class='op'>=</span> <span class='cn'>TRUE</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>pref</th>
      <td><p>Prefix of <em>PLINK/EIGENSTRAT/PACKEDANCESTRYMAP</em> files.
<em>EIGENSTRAT/PACKEDANCESTRYMAP</em> have to end in <code>.geno</code>, <code>.snp</code>, <code>.ind</code>, <em>PLINK</em> has to end in <code>.bed</code>, <code>.bim</code>, <code>.fam</code></p></td>
    </tr>
    <tr>
      <th>outdir</th>
      <td><p>Directory where data will be stored.</p></td>
    </tr>
    <tr>
      <th>inds</th>
      <td><p>Individuals for which data should be extracted</p></td>
    </tr>
    <tr>
      <th>pops</th>
      <td><p>Populations for which data should be extracted. If both <code>pops</code> and <code>inds</code> are provided, they should have the same length and will be matched by position. If only <code>pops</code> is provided, all individuals from the <code>.ind</code> or <code>.fam</code> file in those populations will be extracted. If only <code>inds</code> is provided, each indivdual will be assigned to its own population of the same name. If neither <code>pops</code> nor <code>inds</code> is provided, all individuals and populations in the <code>.ind</code> or <code>.fam</code> file will be extracted.</p></td>
    </tr>
    <tr>
      <th>blgsize</th>
      <td><p>SNP block size in Morgan. Default is 0.05 (50 cM). If <code>blgsize</code> is 100 or greater, if will be interpreted as base pair distance rather than centimorgan distance.</p></td>
    </tr>
    <tr>
      <th>cols_per_chunk</th>
      <td><p>Number of populations per chunk. Lowering this number will lower the memory requirements when running</p></td>
    </tr>
    <tr>
      <th>maxmiss</th>
      <td><p>Discard SNPs which are missing in a fraction of populations higher than <code>maxmiss</code></p></td>
    </tr>
    <tr>
      <th>minmaf</th>
      <td><p>Discard SNPs with minor allele frequency less than <code>minmaf</code></p></td>
    </tr>
    <tr>
      <th>maxmaf</th>
      <td><p>Discard SNPs with minor allele frequency greater than than <code>maxmaf</code></p></td>
    </tr>
    <tr>
      <th>minac2</th>
      <td><p>Discard SNPs with allele count lower than 2 in any population (default <code>FALSE</code>). This option should be set to <code>TRUE</code> when computing f3-statistics where one population consists mostly of pseudohaploid samples. Otherwise heterozygosity estimates and thus f3-estimates can be biased. <code>minac2 == 2</code> will discard SNPs with allele count lower than 2 in any non-singleton population (this option is experimental and is based on the hypothesis that using SNPs with allele count lower than 2 only leads to biases in non-singleton populations). While the <code>minac2</code> option discards SNPs with allele count lower than 2 in any population, the <code><a href='qp3pop.html'>qp3pop</a></code> function will only discard SNPs with allele count lower than 2 in the first (target) population (when the first argument is the prefix of a genotype file).</p></td>
    </tr>
    <tr>
      <th>outpop</th>
      <td><p>Keep only SNPs which are heterozygous in this population</p></td>
    </tr>
    <tr>
      <th>outpop_scale</th>
      <td><p>Scale f2-statistics by the inverse <code>outpop</code> heteroygosity (<code>1/(p*(1-p))</code>). Providing <code>outpop</code> and setting <code>outpop_scale</code> to <code>TRUE</code> will give the same results as the original <em>qpGraph</em> when the <code>outpop</code> parameter has been set, but it has the disadvantage of treating one population different from the others. This may limit the use of these f2-statistics for other models.</p></td>
    </tr>
    <tr>
      <th>transitions</th>
      <td><p>Set this to <code>FALSE</code> to exclude transition SNPs</p></td>
    </tr>
    <tr>
      <th>transversions</th>
      <td><p>Set this to <code>FALSE</code> to exclude transversion SNPs</p></td>
    </tr>
    <tr>
      <th>keepsnps</th>
      <td><p>SNP IDs of SNPs to keep. Overrides other SNP filtering options</p></td>
    </tr>
    <tr>
      <th>overwrite</th>
      <td><p>Overwrite existing files in <code>outdir</code></p></td>
    </tr>
    <tr>
      <th>format</th>
      <td><p>Supply this if the prefix can refer to genotype data in different formats
and you want to choose which one to read. Should be <code>plink</code> to read <code>.bed</code>, <code>.bim</code>, <code>.fam</code> files, or <code>eigenstrat</code>, or <code>packedancestrymap</code> to read <code>.geno</code>, <code>.snp</code>, <code>.ind</code> files.</p></td>
    </tr>
    <tr>
      <th>adjust_pseudohaploid</th>
      <td><p>Genotypes of pseudohaploid samples are usually coded as <code>0</code> or <code>2</code>, even though only one allele is observed. <code>adjust_pseudohaploid</code> ensures that the observed allele count increases only by <code>1</code> for each pseudohaploid sample. If <code>TRUE</code> (default), samples that don't have any genotypes coded as <code>1</code> among the first 1000 SNPs are automatically identified as pseudohaploid. This leads to slightly more accurate estimates of f-statistics. Setting this parameter to <code>FALSE</code> treats all samples as diploid and is equivalent to the <em>ADMIXTOOLS</em> <code>inbreed: NO</code> option. Setting <code>adjust_pseudohaploid</code> to an integer <code>n</code> will check the first <code>n</code> SNPs instead of the first 1000 SNPs.</p></td>
    </tr>
    <tr>
      <th>afprod</th>
      <td><p>Write files with allele frequency products for every population pair. Setting this to FALSE can make <code>extract_f2</code> faster and will require less memory.</p></td>
    </tr>
    <tr>
      <th>fst</th>
      <td><p>Write files with pairwise FST for every population pair. Setting this to FALSE can make <code>extract_f2</code> faster and will require less memory.</p></td>
    </tr>
    <tr>
      <th>poly_only</th>
      <td><p>Specify whether SNPs with identical allele frequencies in every population should be discarded (<code>poly_only = TRUE</code>), or whether they should be used (<code>poly_only = FALSE</code>). By default (<code>poly_only = c("f2")</code>), these SNPs will be used to compute FST and allele frequency products, but not to compute f2 (this is the default option in the original ADMIXTOOLS).</p></td>
    </tr>
    <tr>
      <th>apply_corr</th>
      <td><p>Apply small-sample-size correction when computing f2-statistics (default <code>TRUE</code>)</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Print progress updates</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>SNP metadata (invisibly)</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p><code>extract_f2_large</code> requires less memory because it writes allele frequency data to disk, and doesn't store the allele frequency matrix for all populations and SNPs in memory. If you still run out of memory, reduce <code>cols_per_chunk</code>. This function is a wrapper around <code><a href='extract_afs.html'>extract_afs</a></code> and <code><a href='afs_to_f2.html'>afs_to_f2</a></code>, and is slower than <code><a href='extract_f2.html'>extract_f2</a></code>. It may be faster to call <code><a href='extract_afs.html'>extract_afs</a></code> and <code><a href='afs_to_f2.html'>afs_to_f2</a></code> directly, parallelizing over the different calls to <code><a href='afs_to_f2.html'>afs_to_f2</a></code>.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='extract_f2.html'>extract_f2</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='va'>pref</span> <span class='op'>=</span> <span class='st'>'my/genofiles/prefix'</span>
<span class='va'>f2dir</span> <span class='op'>=</span> <span class='st'>'my/f2dir/'</span>
<span class='fu'>extract_f2_large</span><span class='op'>(</span><span class='va'>pref</span>, <span class='va'>f2dir</span>, pops <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'popA'</span>, <span class='st'>'popB'</span>, <span class='st'>'popC'</span><span class='op'>)</span><span class='op'>)</span>
<span class='op'>}</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Robert Maier, Nick Patterson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: 'c30b6be96312414d83a0ba52b94b558b',
    indexName: 'admixtools',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


